apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs

```diff
@@ -27,8 +27,15 @@ public sealed class Settings
     {
         /// <summary>
         /// Flow control settings.
-        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>. 
+        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>.
         /// </summary>
+        /// <remarks>
+        /// Note that flow control is applied on a per-<see cref="SubscriberServiceApiClient"/> basis: a single
```

Comment:
> `/// Note that flow control is applied on a per-<see cref="SubscriberServiceApiClient"/> basis`

Well it is and it isn't, which is somewhat confusing.

Flow control is applied on a per SubscriberServiceApiClient basis for limiting the pulled messages. But it is applied on a per SubscriberClient basis for message processing. This means that there not only may be more outstanding messages than expected (up to clientCount*maxOutstandingMessages) but also that we are not processing concurrently more than maxOutstandingMessages. I think this disparity is what's most confusing (including for #10358).

Reply:
Righto - will chat with you more tomorrow and try to expand this.

Reply:
Okay, I've had another go at this - PTAL.